inference-engine/src/vpu/common/src/ngraph/transformations/merge_gather_gather_elements.cpp

```diff
@@ -0,0 +1,156 @@
+    for (const auto& node : f->get_ordered_ops()) {
+        if (!m->match(node)) {
+            continue;
+        }
+
+        auto& patternMap = m->get_pattern_value_map();
+
+        std::vector<ngraph::Node*> shapeOfs;
+        std::vector<ngraph::Node*> gatherElements;
+
+        const auto& matchedTranspose = patternMap.at(transpose);
+        const auto& transposeConsumers = matchedTranspose.get_target_inputs();
+
+        const auto isShapeOfOrGatherElements = [](const ngraph::Input<ngraph::Node>& consumer) {
+            return consumer.get_node()->get_type_info() == ngraph::opset6::ShapeOf::type_info ||
+                   consumer.get_node()->get_type_info() == ngraph::opset6::GatherElements::type_info;
+        };
+
+        if (!std::all_of(transposeConsumers.cbegin(), transposeConsumers.cend(), isShapeOfOrGatherElements)) {
+            continue;
+        }
+
+        for (const auto& transposeConsumer : transposeConsumers) {
+            if (transposeConsumer.get_node()->get_type_info() == ngraph::opset6::ShapeOf::type_info) {
+                shapeOfs.push_back(transposeConsumer.get_node());
+            } else if (transposeConsumer.get_node()->get_type_info() == ngraph::opset6::GatherElements::type_info) {
+                gatherElements.push_back(transposeConsumer.get_node());
+            }
+        }
+
+        const auto& matchedTransposePerm = patternMap.at(transposePerm);
+        const auto& matchedSqueezeAxes = patternMap.at(squeezeAxes);
+        const auto& matchedSqueeze = patternMap.at(squeeze);
+        const auto& matchedGatherData = patternMap.at(gatherData);
+        const auto& matchedGatherIndices = patternMap.at(gatherIndices);
+        const auto& matchedGather = patternMap.at(gather);
+
+        for (const auto& gatherElement : gatherElements) {
+            const auto transposeIndices = std::make_shared<ngraph::opset6::Transpose>(gatherElement->input_value(1), matchedTransposePerm);
+            const auto unsqueezeIndices = std::make_shared<ngraph::opset6::Unsqueeze>(
+                transposeIndices,
+                matchedSqueezeAxes);
+            const auto expGatherElements = std::make_shared<ngraph::vpu::op::ExpGatherElements>(
+                matchedGatherData,
+                unsqueezeIndices,
+                matchedGatherIndices,
+                ngraph::as_type<ngraph::opset6::GatherElements>(gatherElement)->get_axis(),
+                ngraph::as_type<ngraph::opset6::Gather>(matchedGather.get_node())->get_axis());
+            const auto squeezeData = matchedSqueeze.get_node()->clone_with_new_inputs({expGatherElements, matchedSqueezeAxes});
+            const auto transposeData = matchedTranspose.get_node()->clone_with_new_inputs({squeezeData, matchedTransposePerm});
+            transposeData->set_friendly_name(gatherElement->get_friendly_name());
+            gatherElement->get_default_output().replace(transposeData);
```

Comment:
Use `replace_node` or specify output port explicitly as `get_default_output` is smth that will be deprecated soon.

Reply:
Specified the port explicitly.